lucene/core/src/java/org/apache/lucene/search/similarities/TFSimilarity.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search.similarities;
+
+import org.apache.lucene.index.FieldInvertState;
+import org.apache.lucene.search.CollectionStatistics;
+import org.apache.lucene.search.TermStatistics;
+
+/** Similarity that returns the TF as score */
+public class TFSimilarity extends Similarity {
+
+  @Override
+  public long computeNorm(FieldInvertState state) {
+    return 1; // we dont care
+  }
+
+  @Override
+  public SimScorer scorer(
+      float boost, CollectionStatistics collectionStats, TermStatistics... termStats) {
+    return new SimScorer() {
+      @Override
+      public float score(float freq, long norm) {
+        return freq;
```

Comment:
Should this be `boost * freq`?

Reply:
Good catch, thanks! Will also add test coverage to match, marking as draft in the meantime.
```suggestion
        return boost * freq;
```